### PR TITLE
Freezed shuffled test splits & added xsum-test data subset #56

### PR DESCRIPTION
**Pull request by anton164**
- Shuffled split ids for each data subset [xsum-test, xent-test, test-extrinsic]
  are sampled once and written to "data/xsum_shuffled_test_splits.json"

- Load from shuffled test split loads first N ids according to test_size argument

### CODE REVIEW

File: evaluate_summaries.py [hunk 1]
```diff
@@ -31,19 +30,13 @@
...
         )
-        test_set_ids = {k for (k, v) in test_set}
     else:
-        test_set = (
-            load_extrinsic_test_set(
-                xsum_test, baseline_metadata, gold_metadata, args.test_size
-            )
-            if args.data_subset == "test-extrinsic"
-            else load_xent_test_set(xsum_test, gold_metadata, args.test_size)
+        test_set_ids = set(
```
Comment (anton164) [inline evaluate_summaries.py:37]: I verified that this change works as intended by running evaluate_summaries with test-size 100 (it yields the same results).

File: construct_shuffled_test_splits.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def filter_xsum(sum_id):
+        return "xent-train" not in gold_metadata[sum_id]
+
+    def filter_xent_test(sum_id):
+        return "xent-test" in gold_metadata[sum_id]
+
+    def filter_xent_extrinsic(sum_id):
+        return (
+            "xent-train" not in gold_metadata[sum_id]
```
Comment (dleve123) [inline construct_shuffled_test_splits.py:31]: in addition to ensuring that the example is not in `xent-train`, shouldn't we make sure that the example is within `xent-test?` At least, that's what I would presume from a function name of `filter_xent_extrinsic`

Comment (dleve123) [reply construct_shuffled_test_splits.py:31]: or maybe we should rename the function -- if i remember correctly, this is our legacy test set, which might include examples included in xent-test or not (but all are in xsum-test)

**Review by dleve123 — COMMENTED** [review]
generally looks good - left some questions, see suggested function rename, in particular, @anton164

File: construct_shuffled_test_splits.py (hunk 2 above)
Comment (anton164) [reply construct_shuffled_test_splits.py:31]: My bad, this function name is incorrect - it should be `filter_xsum_extrinsic`

**Pull request merged**